ftree/node.go

```diff
@@ -0,0 +1,59 @@
+package ftree
+
+import (
+	"fmt"
+	"sort"
+)
+
+type Node map[string]Node
+
+// Print recurses through n and its member Nodes to pretty-print a Node as a
+// tree such as:
+//
+//     ├── ftree
+//     │   ├── node.go
+//     │   ├── tree.go
+//     │   └── tree_test.go
+//     └── main.go
+//
+// It should initially be called with as n.Print(0, nil) as the arguments are
+// primarily interesting during recursion.
+func (n Node) Print(indent int, leaders []string) {
+	var keys []string
+	for k := range n {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for i, part := range keys {
```

Comment:
`keys` is a slice (array) of the keys from the Node (which is a map). In the example above, they'd be `[]string{"ftree", "main.go"}`

That exists because I want to sort the keys, and the order of iterating over the map isn't deterministic or sorted.

On this line, `i` is the iteration count. `part`, a terrible name,  is a "part" of `ftree/node.go` for example. It could be `ftree` here.

All that to say, I am open to different names. `keys` and `part` actually did have some thought behind them, but they are admittedly confusing.